core/services/job/orm.go

```diff
@@ -129,50 +139,76 @@ func (o *orm) CreateJob(ctx context.Context, jobSpec *models.JobSpecV2, taskDAG
 }
 
 // DeleteJob removes a job that is claimed by this orm
-// TODO: Extend this in future so it can delete any job and other nodes handle
-// it gracefully
-// See: https://www.pivotaltracker.com/story/show/175287919
 func (o *orm) DeleteJob(ctx context.Context, id int32) error {
 	o.claimedJobsMu.Lock()
 	defer o.claimedJobsMu.Unlock()
 
-	idx := -1
-	for i, j := range o.claimedJobs {
-		if j.ID == id {
-			idx = i
-			break
-		}
+	err := o.db.Exec(`
+            WITH deleted_jobs AS (
```

Comment:
Yes we considered it and went back and forth a fair bit, there was a good reason for doing the fk this way round which eludes me at the moment.

Reply:
@spooktheducks do you remember?

Reply:
fwiw I had a very similar debate in the recent past and we went with the other way and it scaled beautifully up to like 20 child tables

Reply:
Unfortunately I don't -- that sprint was a complete blur

Reply:
I remember why - because it makes no sense to ever have a job without a spec, but it might make sense to have a spec without a job.

Hence the FK should belong on the job otherwise you could have orphaned jobs.

Reply:
I see. Given we already have the separation of jobs and runs, what would be a scenario where you have a spec without a job?

Reply:
Also I guess we plan have nullable FKs in the jobs table when we add new job types?

Reply:
> nullable FKs in the jobs table

That's the plan, yeah.

Reply:
@connorwstein The idea is to modify the check constraint to enforce exactly one foreign key.